src/modules/launcher/Plugins/Microsoft.Plugin.Indexer/SearchHelper/WindowsSearchAPI.cs

```diff
@@ -32,21 +36,21 @@ public List<SearchResult> ExecuteQuery(ISearchQueryHelper queryHelper, string ke
             // Loop over all records from the database
             foreach (OleDBResult oleDBResult in oleDBResults)
             {
-                if (oleDBResult.fieldData[0] == DBNull.Value || oleDBResult.fieldData[1] == DBNull.Value || oleDBResult.fieldData[2] == DBNull.Value)
+                if (oleDBResult.FieldData[0] == DBNull.Value || oleDBResult.FieldData[1] == DBNull.Value || oleDBResult.FieldData[2] == DBNull.Value)
```

Comment:
while not you, we need to document the result magic numbers here

Reply:
@alekhyareddy28.   @crutkas has a good point here.  Not only are we using magic numbers, but I think we a leaking a bunch of information that could be better encapsulated in the OleDBResult.   I started down the following route, but realized this should be it's own thread of work.  Let me know what you think and we can create a new techdebt item for this. 
1.  Rename OleDBResult to something that indicates the type of result we expect "IndexedItemResult". 
2. Enforce the class invariants at construction time.  Basically throwing if fieldData is null or if fieldData indexes 0,1,2 are null. 
3. Expose public Fields for UriPath, Title, and FileAttributes. 

Note:  I'm trying to make it so that a "OleDBResult' cannot be constructed in an invalid state and then prevents having check IsValid.  This then would force the OleDBSearch to enforce that it doesn't attempt to construct an invalid OleDBSearch field.    The problem here is that OleDBSearch is actually more generic than the OleDBResult that I would expect to return.  We need to inject some construct (eg. IOleDBResultBuilder) that knows how to build a result and verify if it is valid.  

The above is a bit philisophical so I'd love to hear your thoughts, and we can create a new issue to be triaged for future work.